components/forum/DiscourseAdapter.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace app\components\forum;
+
+use app\models\User;
+use yii\base\Component;
+use yii\base\InvalidConfigException;
+use yii\db\Connection;
+use yii\db\Query;
+use yii\di\Instance;
+use yii\helpers\ArrayHelper;
+use yii\helpers\Inflector;
+use yii\httpclient\Client;
+
+/**
+ * DiscourseAdapter implements a forum bridge between the Discourse Forum and the application.
+ * Configure as follows:
+ *
+ * ```php
+ * 'forumBridge' => [
+ *      'class' => \app\components\forum\DiscourseAdapter::class,
+ *      'apiUrl' => 'https://forum.yiiframework.com/',
+ *      'apiToken' => '123456',
+ *  ],
+ * ```
+ *
+ * @see https://docs.discourse.org/
+ */
+class DiscourseAdapter extends Component implements ForumAdapterInterface
+{
+    /**
+     * @var string discourse API URL.
+     */
+    public $apiUrl;
+    /**
+     * @var string discourse API auth token.
+     */
+    public $apiToken;
+
+
+    /**
+     * @return Client
+     */
+    private function getClient()
+    {
+        $client = new Client([
+            'baseUrl' => $this->apiUrl,
+        ]);
+
+        return $client;
+    }
+
+
+    public function getPostDate($user, $number)
+    {
+        if (!$user->forum_id) {
+            return false;
+        }
+
+        $tablePrefix = $this->tablePrefix;
+        $n = ((int) $number) - 1;
+        $sql = "SELECT post_date FROM {$tablePrefix}posts WHERE author_id = :user_id ORDER BY post_date ASC LIMIT " . ($n > 0 ? "$n," : '') . '1';
```

Comment:
Is there API for the purpose?

Reply:
this is still copy paste from IPB :)

getting this from the API is not that easy though.